### PR TITLE
GSDK-1981 : Fix  Screen Capturer crash targeting Android Q

### PR DESCRIPTION
**Pull request by kbagchiGWC**
Android Q now requires developers to specify a [foreground service](https://developer.android.com/about/versions/10/features#fg-service-types) when using the MediaProjection API. Implemented `ScreenCapturerManager` and `ScreenCapturerService`, that start/stop foreground service before/after screen capture.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerService.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
+package com.twilio.video.examples.screencapturer;
```
Comment (aaalaniz) [inline exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerService.java]: Nit: Can we separate these two classes into two separate files?

File: exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java [hunk 2]
```diff
@@ -49,6 +50,18 @@ public void onCreate(Bundle savedInstanceState) {
         localVideoView = (VideoView) findViewById(R.id.local_video);
     }
 
+    @Override
```
Comment (aaalaniz) [inline exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java]: Can you scope the screen capturer manager to onCreate/onDestroy?

This way the user can continue capturing the screen in the background with a notification visible.

File: exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java [hunk 3]
```diff
@@ -68,7 +81,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.share_screen_menu_item:
                 String shareScreen = getString(R.string.share_screen);
-
+                screenCapturerManager.nextState();
```
Comment (aaalaniz) [inline exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java]: I think the use of the state machine pattern works well within the screen capturer service itself, but I think we should provide a more declarative API such as startService/endService so the user has a clearer understanding of what is happening when the button is pressed.

File: exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerService.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+     * An enum describing the possible states of a ScreenCapturerManager.
+     */
+    public enum State {
+        BIND_SERVICE,
+        START_FOREGROUND,
+        END_FOREGROUND,
+        UNBIND_SERVICE
+    }
+
+    ScreenCapturerManager(Context context) {
```
Comment (aaalaniz) [inline exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerService.java]: One observation here is that the ScreenCapturerManager and Service will be used the same for every API level. What are your thoughts on making the ScreenCapturerManager API only used on Android Q and up? We can do this at the activity level. This ensures we don't have to debug Service management issues on API levels where it is not required to capture the screen.

File: exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java (hunk 3 above)
Comment (kbagchiGWC) [reply exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java]: That's what I initially had .. then moved to a simpler API approach. I can move back to the more declarative API.

File: exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java (hunk 2 above)
Comment (aaalaniz) [reply exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java]: I think you can remove the onStart onStop if they are not be used

**Pull request merged**